packages/rs-drive-abci/src/execution/validation/state_transition/processor/v0/mod.rs

```diff
@@ -532,15 +530,16 @@ impl StateTransitionBasicStructureValidationV0 for StateTransition {
             StateTransition::IdentityCreditTransfer(st) => {
                 st.validate_basic_structure(platform_version)
             }
+            StateTransition::DataContractCreate(st) => {
+                st.validate_basic_structure(platform_version)
+            }
+            StateTransition::DataContractUpdate(st) => {
+                st.validate_basic_structure(platform_version)
+            }
         }
     }
     fn has_basic_structure_validation(&self) -> bool {
-        !matches!(
-            self,
-            StateTransition::DataContractCreate(_)
-                | StateTransition::DataContractUpdate(_)
-                | StateTransition::MasternodeVote(_)
-        )
+        !matches!(self, StateTransition::MasternodeVote(_))
```

Comment:
This won't work, because it's dependent on the platform version. I think we should check the platform version instead to check to see if it has basic structure validation for the version we currently are on.